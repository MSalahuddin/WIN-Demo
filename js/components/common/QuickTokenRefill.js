import React, { useState, useContext, useEffect } from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { SafeAreaView, Platform, FlatList, StyleSheet, View, ImageBackground } from 'react-native';
import LottieView from 'lottie-react-native';
import * as RNIap from 'react-native-iap';
import Text, { SIZE, FONT_FAMILY } from './Text';
import api from '../../api';
import QuickTokenCard from './QuickTokenCard';
import { ExitCircle, howtoBackgroundPopUp } from '../../../assets/images';
import { scaleHeight, scale } from '../../platformUtils';
import { color } from '../../styles';
import IconButton from './IconButton';
import { UserContext } from '../../context/User.context';
import { PopupContext } from '../../context/Popup.context';
import { gamePlayStrings } from '../../stringConstants';
import { quickRefill } from '../../../assets/animations';
import { styles } from './styles';

const StyledModal = styled(Modal)`
align-items: center;
background-color: ${color.popupBlack};
justify-content: center;
margin: 0;
`;

const ContentContainer = styled.ImageBackground`
flex-direction: column;
margin-top : ${scaleHeight(50)}
`;

const TopHeadingContainer = styled.View`
flex-direction: column;
align-self:center;
width:95%;
`;

const TextWrapper = styled(Text)`
margin-vertical : ${scale(10)};
`;

const ButtonText = styled(Text)`
justify-content: center;
border-bottom-color: white;
border-bottom-width: 1px;
`;

const ButtonWrapper = styled.TouchableOpacity`
align-items: center;
`;

const ExitCircleBtn = styled(IconButton)`
position:absolute;
top: -40 ;
position: absolute;
z-index: 1;
right:-10;
`;

const TokenListView = styled.View`
height:${scaleHeight(200)}
`;

const QuickTokenRefill = ({

  isVisible,
  onPress,
  onPressClose
}) => {
  const [tokenPackData, setTokenPackData] = useState(null);
  const { shouldRefreshTokenPack, fetchPoints,isUserLoggedIn } = useContext(UserContext);
  const { displayRequestError,setIsCreateAccountPopUpShown } = useContext(PopupContext);
  const [position, setPosition] = useState(0);
  const renderTokenCard = ({
    imageUrl,
    name,
    tokenAmount,
    isOnSale,
    bonusTokenAmount,
    ribbonColor,
    ribbonName,
    iapData,
    iOSProductId,
    androidProductId
  }) => (
    <QuickTokenCard
      imageUrl={imageUrl}
      name={name}
      isOnSale={isOnSale}
      tokenAmount={tokenAmount}
      bonusTokenAmount={bonusTokenAmount}
      ribbonColor={ribbonColor}
      ribbonName={ribbonName}
      iapData={iapData}
      // eslint-disable-next-line no-use-before-define
      onPress={() => onTokenPurchasePress(Platform.OS === "android" ? androidProductId : iOSProductId)}
    />
  );

  const fetchAppStoreTokens = async tokenPacks => {
    const tokenIds = Platform.OS === 'android' ?
      tokenPacks.map(({ androidProductId }) => {
        if (androidProductId === "")
          return null
        return androidProductId
      }).filter(Boolean) :
      tokenPacks.map(({ iOSProductId }) =>  iOSProductId);
    try {
      const result = await RNIap.getProducts(tokenIds);
      const tokenPacksWithIapData = tokenPacks
        .map(token => {
          const iapData = result.find(({ productId }) => {
            if (Platform.OS === 'android') {
              return productId === token.androidProductId
            }
            return productId === token.iOSProductId;
          });
          return (iapData &&  token?.isQuickRefill)?{ ...token, iapData } : {};
        })
        .filter(pack => pack.iapData);
      setTokenPackData(tokenPacksWithIapData);
    } catch (error) {
      displayRequestError(error.message);
    }
  };

  const fetchTokenPacks = async code => {
    try {
      const res = await api.getTokenPacks(code, Platform.OS === 'android');
      if (res.status === 200 && res.data) {
        await fetchAppStoreTokens(res.data);
      }
    } catch (error) {
      displayRequestError(error.message);
    }
  };

  useEffect(() => {
    // Check if deep link has promoCode meta data
    fetchTokenPacks();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (shouldRefreshTokenPack) {
      // Token pack was successfully purchased - refresh
      fetchTokenPacks();
      fetchPoints();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [shouldRefreshTokenPack]);

  const requestPurchase = async productId => {
    try {
      await RNIap.initConnection()
      await RNIap.requestPurchase(productId, false);
      onPressClose()

    } catch (error) {
      onPressClose()
      // Error handled in purchase listener
    }
  };

  const onTokenPurchasePress = iOSProductId => {
    // if (!isUserLoggedIn) {
    //   setIsCreateAccountPopUpShown(true);
    // } else {
    requestPurchase(iOSProductId);
    // }


  };

  return (
    <SafeAreaView>
      <StyledModal isVisible={isVisible}>
        <View style={styles.mainView}>
          <ImageBackground
            source={howtoBackgroundPopUp}
            style={styles.modalImg}
            resizeMode={'stretch'} >

            <ExitCircleBtn
              testID="exit-circle"
              onPress={onPressClose}
              icon={ExitCircle}
              size={40}
            />

            <View style={styles.refillIcon} >
              <LottieView source={quickRefill} autoPlay loop resizeMode="cover" />
            </View>

            <ContentContainer>
              <TopHeadingContainer>
                <TextWrapper
                  size={SIZE.XXXLARGE}
                  fontFamily={FONT_FAMILY.BOLD}
                  color={color.white}
                  alignCenter
                >
                  {gamePlayStrings.quickRefill}
                </TextWrapper>
                <TextWrapper
                  size={SIZE.XSMALL}
                  fontFamily={FONT_FAMILY.BOLD}
                  color={color.white}
                  alignCenter
                >
                  {gamePlayStrings.RefillString}
                </TextWrapper>
              </TopHeadingContainer>
              <TokenListView>
                <FlatList
                  contentContainerStyle={styles.scrollViewContainerStyle}
                  data={tokenPackData}
                  renderItem={({ item }) => renderTokenCard(item)}
                  keyExtractor={item => item.name}
                  numColumns={1}
                  // columnWrapperStyle={flatListStyles.columnStyle}
                  initialNumToRender={4}
                />
              </TokenListView>

              <ButtonWrapper onPress={onPress} >
                <ButtonText width={350} color={color.white} size={SIZE.XXSMALL}>
                  {gamePlayStrings.headToTokenStore}
                </ButtonText>
              </ButtonWrapper>
            </ContentContainer>
          </ImageBackground>
        </View>
      </StyledModal>
    </SafeAreaView>
  );
};

QuickTokenRefill.defaultProps = {
  onPress: () => { },
  onPressClose: () => { }
};

QuickTokenRefill.propTypes = {
  onPress: () => { },
  onPressClose: () => { },
  isVisible: PropTypes.bool.isRequired
};

export default QuickTokenRefill;
