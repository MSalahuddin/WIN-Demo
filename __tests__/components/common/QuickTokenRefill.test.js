import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import MockProvider from '../../MockProvider';
import QuickTokenRefill from '../../../js/components/common/QuickTokenRefill';



describe('Component QuickTokenRefill', () => {
    const onPress = jest.fn();
    const onPressClose = jest.fn();
    test('renders correctly when isUserLoggedIn', () => {
        const output = render(
            <MockProvider>
                <QuickTokenRefill
                    isVisible
                    onPress={onPress}
                    onPressClose={onPressClose}
                />
            </MockProvider>
        );
        expect(output).toMatchSnapshot();
    });

    test('renders correctly when isVisible={false}', () => {
        const output = render(
            <MockProvider>
                <QuickTokenRefill isVisible={false}
                    onPress={onPress}
                    onPressClose={onPressClose}
                />
            </MockProvider>
        );
        expect(output).toMatchSnapshot();
    });

    test('renders correctly when onPress', () => {
        const {getByTestId} = render(
            <MockProvider>
                <QuickTokenRefill
                    isVisible
                    onPress={onPress}
                    onPressClose={onPressClose}
                />
            </MockProvider>
        );
        fireEvent(getByTestId('quick-token-store-button'), 'press');
      expect(onPress).toHaveBeenCalled();
    });
    test('renders correctly when onPressClose', () => {
        const {getByTestId} = render(
            <MockProvider>
                <QuickTokenRefill
                    isVisible
                    onPress={onPress}
                    onPressClose={onPressClose}
                />
            </MockProvider>
        );
        fireEvent(getByTestId('exit-circle'), 'press');
        expect(onPressClose).toHaveBeenCalled();
    });
});
