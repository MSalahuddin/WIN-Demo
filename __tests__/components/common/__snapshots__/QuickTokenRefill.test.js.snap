// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component QuickTokenRefill renders correctly when isUserLoggedIn 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 12,
            "left": 12,
            "right": 12,
            "top": 12,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": 24,
            "marginTop": 48,
            "opacity": 1,
            "position": "absolute",
            "right": 30,
            "top": -40,
            "zIndex": 1,
          }
        }
        testID="exit-circle"
      >
        <Image
          opacity={1}
          resizeMode="contain"
          size={30}
          source={1}
          style={
            Array [
              Object {
                "height": 45,
                "width": 45,
              },
            ]
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            4283239667,
            4283047829,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        onLayout={[Function]}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#FFFFFF",
              "borderRadius": 15,
              "borderWidth": 5,
              "marginTop": 40,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "85%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "marginVertical": 60,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "width": "95%",
                },
              ]
            }
          >
            <Text
              alignCenter={true}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              maxFontSizeMultiplier={1.3}
              size="XXLARGE"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 48,
                    "justifyContent": "center",
                    "textAlign": "center",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginVertical": 15,
                  },
                ]
              }
            >
              QUICK TOKEN REFILL
            </Text>
            <Text
              alignCenter={true}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "center",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginVertical": 15,
                  },
                ]
              }
            >
              Refill your tokens without losing your spot in line!
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 600,
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "paddingBottom": 32,
                }
              }
              data={null}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={4}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="BOLD"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "CalibreBold",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginVertical": 15,
                },
              ]
            }
          >
            Want to purchase a different pack?
          </Text>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(96,202,142,1)",
                  "borderRadius": 88,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 0,
                  "opacity": 1,
                  "width": 500,
                }
              }
              testID="quick-token-store-button"
            >
              <View
                borderRadius={44}
                colorScheme="rgba(128,229,173,1)"
                height={50}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(128,229,173,1)",
                      "borderRadius": 88,
                      "height": 90,
                      "justifyContent": "center",
                      "width": 494,
                    },
                  ]
                }
                width={250}
              >
                <View
                  borderRadius={44}
                  colorScheme="rgba(114,213,158,1)"
                  height={50}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(114,213,158,1)",
                        "borderRadius": 108,
                        "height": 86,
                        "justifyContent": "center",
                        "marginTop": 8,
                        "width": 494,
                      },
                    ]
                  }
                  width={250}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      alignCenter={false}
                      allowFontScaling={true}
                      color="#FFFFFF"
                      fontFamily="MEDIUM"
                      isUnderlined={false}
                      isUppercase={false}
                      maxFontSizeMultiplier={1.3}
                      size="NORMAL"
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontFamily": "Calibre-Medium",
                            "fontSize": 30,
                            "justifyContent": "center",
                            "textAlign": "left",
                            "textDecorationColor": "#FFFFFF",
                            "textDecorationLine": "none",
                            "textDecorationStyle": "solid",
                            "textTransform": "none",
                          },
                          Object {
                            "marginTop": 14,
                          },
                        ]
                      }
                    >
                      Head To Token Store
                    </Text>
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "height": 40,
                            "marginLeft": 7.5,
                            "width": 40,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </Modal>
</RCTSafeAreaView>
`;

exports[`Component QuickTokenRefill renders correctly when isVisible={false} 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 12,
            "left": 12,
            "right": 12,
            "top": 12,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": 24,
            "marginTop": 48,
            "opacity": 1,
            "position": "absolute",
            "right": 30,
            "top": -40,
            "zIndex": 1,
          }
        }
        testID="exit-circle"
      >
        <Image
          opacity={1}
          resizeMode="contain"
          size={30}
          source={1}
          style={
            Array [
              Object {
                "height": 45,
                "width": 45,
              },
            ]
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            4283239667,
            4283047829,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        onLayout={[Function]}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#FFFFFF",
              "borderRadius": 15,
              "borderWidth": 5,
              "marginTop": 40,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "85%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "marginVertical": 60,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "width": "95%",
                },
              ]
            }
          >
            <Text
              alignCenter={true}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              maxFontSizeMultiplier={1.3}
              size="XXLARGE"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 48,
                    "justifyContent": "center",
                    "textAlign": "center",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginVertical": 15,
                  },
                ]
              }
            >
              QUICK TOKEN REFILL
            </Text>
            <Text
              alignCenter={true}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "center",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginVertical": 15,
                  },
                ]
              }
            >
              Refill your tokens without losing your spot in line!
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 600,
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "paddingBottom": 32,
                }
              }
              data={null}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={4}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="BOLD"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "CalibreBold",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginVertical": 15,
                },
              ]
            }
          >
            Want to purchase a different pack?
          </Text>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(96,202,142,1)",
                  "borderRadius": 88,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 0,
                  "opacity": 1,
                  "width": 500,
                }
              }
              testID="quick-token-store-button"
            >
              <View
                borderRadius={44}
                colorScheme="rgba(128,229,173,1)"
                height={50}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(128,229,173,1)",
                      "borderRadius": 88,
                      "height": 90,
                      "justifyContent": "center",
                      "width": 494,
                    },
                  ]
                }
                width={250}
              >
                <View
                  borderRadius={44}
                  colorScheme="rgba(114,213,158,1)"
                  height={50}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(114,213,158,1)",
                        "borderRadius": 108,
                        "height": 86,
                        "justifyContent": "center",
                        "marginTop": 8,
                        "width": 494,
                      },
                    ]
                  }
                  width={250}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      alignCenter={false}
                      allowFontScaling={true}
                      color="#FFFFFF"
                      fontFamily="MEDIUM"
                      isUnderlined={false}
                      isUppercase={false}
                      maxFontSizeMultiplier={1.3}
                      size="NORMAL"
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontFamily": "Calibre-Medium",
                            "fontSize": 30,
                            "justifyContent": "center",
                            "textAlign": "left",
                            "textDecorationColor": "#FFFFFF",
                            "textDecorationLine": "none",
                            "textDecorationStyle": "solid",
                            "textTransform": "none",
                          },
                          Object {
                            "marginTop": 14,
                          },
                        ]
                      }
                    >
                      Head To Token Store
                    </Text>
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "height": 40,
                            "marginLeft": 7.5,
                            "width": 40,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
